src/fastoad/module_management/service_registry.py

```diff
@@ -486,3 +486,10 @@ def get_submodel(cls, service_id: str, options: dict = None):
             instance = om.Group()
 
         return instance
+
+    @classmethod
+    def cancel_submodel_deactivations(cls):
+        """Reactivates all submodels that have been deactivated."""
+        deactivated_submodel_ids = [k for k, v in cls.active_models.items() if not v]
+        for submodel_id in deactivated_submodel_ids:
+            del cls.active_models[submodel_id]
```

Comment:
Out of curiosity, why does deleting the deactivated submodels reactivate them ?

Reply:
The `RegisterSubmodel.active_models` class attribute is by default an empty dict. When a submodel is required, `RegisterSubmodel.active_models` is checked for the requirement ID. If it is found, then a choice has been expressed and the dict value is used (a submodel ID or `null`). If it is not found, then it means that no choice has been made, so either there is only one matching submodel, or an error will be raised.

Therefore, by removing the submodel ID from `RegisterSubmodel.active_models` keys, we simply get back to the no-choice state.